static/space/js/Observers.js

```diff
@@ -11,11 +11,9 @@ export default class Observers {
     const observers = this.getObserverList(key);
     observers.push(cb);
 
-    const unloadListener = () => {
+    win.addEventListener('unload', () => {
       observers.splice(observers.indexOf(cb), 1);
-      win.removeEventListener('unload', unloadListener);
-    };
-    win.addEventListener('unload', unloadListener);
+    }, { once: true });
```

Comment:
I believe this is functionally the same as what you had? Add a listener that's removed when it triggers?